omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,144 @@
+from math import sqrt
+from functools import reduce
+from operator import mul
+
+import tfs
+from tfs.tools import significant_digits
+
+from omc3.utils.logging_tools import get_logger
+from omc3.definitions.constants import PLANES
+from generic_parser import entrypoint, EntryPointParameters
+
+LOG = get_logger(__name__)
+
+BETASTAR = 'BETSTAR'
+ERR = 'ERR'
+LABEL = 'LABEL'  # Column containing the IP/Beam names
+BEAMS = ('B1', 'B2')
+IPS = ('IP1', 'IP5')
+LABELS = [f'{i}{b}' for i in IPS for b in BEAMS]
+
+
+def get_params():
+    return EntryPointParameters(
+            tfs=dict(
+                required=True,
+                type=str,
+                help="TFS file containing the β* and error for "
+                     "B{1,2}{H,V}IP{1,5}"
+            ),
+            inplace=dict(
+                required=False,
+                action='store_true',
+                help="Add the luminosity imbalance in the source TFS file as a"
+                     "header"
+            )
+    )
+
+
+def _validate_tfs(df):
+    # First validate the labels, which correspond to the IP and Beam
+    expected_labels = set(LABELS)
+    if len(df[LABEL]) != 4 or set(df[LABEL]) != expected_labels:
+        for label in LABELS:
+            if len(df.loc[df[LABEL] == label]) > 1:
+                msg = f'Found label {label} several times. Expected only once.'
+                LOG.error(msg)
+                raise KeyError(msg)
+
+        msg = 'Expected labels in the TFS file not found. Expected labels:' \
+              f' {expected_labels}'
+        LOG.error(msg)
+        raise KeyError(msg)
+
+    # Validate the columns we need: BETSTAR{X,Y} and ERRBETSTAR{X,Y}
+    expected_columns = [f'{BETASTAR}{p}' for p in PLANES] + \
+                       [f'{ERR}{BETASTAR}{p}' for p in PLANES]
+    if not all([column in df.columns for column in expected_columns]):
+        msg = 'Expected columns in the TFS file not found. Expected ' \
+              f'columns: {expected_columns}'
+        LOG.error(msg)
+        raise KeyError(msg)
+
+
+def get_imbalance(df):
+    rel_error = {}
+    eff_beta = {}
+    for ip in IPS:
+        # Get the individual rows for the IP1 and IP5
+        row = df.loc[df[LABEL].str.startswith(ip)]
+
+        # Compute the average β* for each plane with both beams
+        # and the absolute errors for each plane
+        average_beta = {}
+        error = {}
+        for p in PLANES:
+            average_beta[p] = row.sum(axis=0)[f'{BETASTAR}{p}'] / 2
+            error[p] = 0.5 * sqrt((row[f'{ERR}{BETASTAR}{p}'] ** 2).sum(axis=0))
+
+        # Compute the relative error for this IP
+        rel_error[ip] = 0
+        for p in PLANES:
+            rel_error[ip] += (error[p] ** 2) / (average_beta[p] ** 2)
+        rel_error[ip] = 0.5 * sqrt(rel_error[ip])
```

Comment:
`rel_error[ip] = 0.5 * sqrt(sum([(error[p] ** 2) / (average_beta[p] ** 2) for p in planes]))`

if you like